lsp-mode.el

```diff
@@ -404,5 +404,27 @@ If WORKSPACE is not specified defaults to lsp--cur-workspace."
       (view-mode 1))
     (switch-to-buffer buffer-name)))
 
+(defun lsp-get-renderer (language)
+  "Return a function to fontify a string in LANGUAGE."
+  (thread-last lsp--cur-workspace
+    lsp--workspace-client
+    lsp--client-string-renderers
+    (assoc-string language)
+    cdr))
+
+(defun lsp-workspace-root (&optional _path)
+  "Find the workspace root for the current file."
+  (when lsp--cur-workspace
+    (lsp--workspace-root lsp--cur-workspace)))
```

Comment:
The pattern ` (lsp--workspace-root lsp--cur-workspace)` has been used 4 times in lsp-mode and twice in lsp-ui. Making a dedicated function sounds good but I'm not sure if the `when` check is necessary

Reply:
I wanted to preserve the original behaviour - there is one place in which the presence of `lsp--cur-workspace` is performed before getting the root - refer to https://github.com/emacs-lsp/lsp-ui/pull/192/commits/43ab6fceeea0eb241d3a80e8da6d248ab9a87097